lib/core/presentation/pages/chat/chat_list/views/widgets/guild_roles_list.dart

```diff
@@ -0,0 +1,162 @@
+  const GuildRolesList({super.key, required this.guildRoom});
+
+  final GuildRoom guildRoom;
+
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+
+    return Column(
+      children: [
+        BlocBuilder<CheckGuildRoomRolesBloc, CheckGuildRoomRolesState>(
+          builder: (context, state) {
+            return state.maybeWhen(
+              success: (guild, guildRoles, guildRolePermissions) {
+                // If didn't select any roles
+                if (guildRoom.guildRoleIds == null) {
+                  if (guild != null) {
+                    return _buildGuildDetailItem(
+                      guild: guild,
+                      context: context,
+                    );
+                  }
+                  return const SizedBox();
+                } else {
+                  return _buildGuildRolesList(
+                    guildRoles,
+                    guildRolePermissions,
+                    guild,
+                    colorScheme,
+                    context,
+                  );
+                }
+              },
+              loading: () => Center(
+                child: Padding(
+                  padding: EdgeInsets.symmetric(vertical: Spacing.medium),
+                  child: Loading.defaultLoading(context),
+                ),
+              ),
+              orElse: () => Center(
+                child: Text(t.common.somethingWrong),
+              ),
+            );
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _buildGuildDetailItem({
+    required BuildContext context,
+    required Guild guild,
+  }) {
+    return Column(
+      children: [
+        SizedBox(height: Spacing.medium),
+        GuildDetailItem(
+          guild: guild,
+          onTap: () {
+            AutoRouter.of(context).navigate(
+              WebviewRoute(
+                uri: Uri.parse(
+                  GuildUtils.getFullGuildUrl(guild.urlName ?? ''),
+                ),
+              ),
+            );
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _buildGuildRolesList(
+    List<GuildRole>? guildRoles,
+    List<GuildRolePermission>? guildRolePermissions,
+    Guild? guild,
+    ColorScheme colorScheme,
+    BuildContext context,
+  ) {
+    final percentageProgress =
+        _calculatePercentageProgress(guildRolePermissions);
+    return Column(
+      children: [
+        SizedBox(height: Spacing.medium),
+        LinearProgressIndicator(
+          value: percentageProgress,
+          color: LemonColor.paleViolet,
+          backgroundColor: colorScheme.surface,
+          minHeight: 3,
+          borderRadius: BorderRadius.circular(LemonRadius.extraSmall),
+        ),
+        SizedBox(height: Spacing.medium),
+        ConstrainedBox(
+          constraints: const BoxConstraints(maxHeight: 300),
+          child: ListView.separated(
+            shrinkWrap: true,
```

Comment:
if the items length is not large then it's ok to use shrinkWrap here, otherwise, please consider

Reply:
in this case, should we consider using CustomScrollView ?